RequestService/RequestService.Repo/Repository.cs

```diff
@@ -2098,62 +2098,59 @@ public async Task<int> OpenJobCount(IEnumerable<int> groups)
                 );
         }
 
-        public async Task<Chart> GetActivitiesByMonth(int groupId)
+        public async Task<List<DataItem>> GetActivitiesByMonth(int groupId, DateTime minDate)
         {
             DateTime dt = DateTime.UtcNow.Date.AddYears(-1);
-            Chart result = new Chart()
-            {
-                Title = "Activities by month",
-                XAxisName = "Month",
-                YAxisName = "Count",
-                ChartType = ChartTypes.Bar,
-                ChartItems = new List<ChartItem>()
-            };
 
-            var chartItems = _context.Job
-                .Include(i => i.NewRequest)
-                .Where(x => x.NewRequest.ReferringGroupId == groupId && x.NewRequest.DateRequested > dt)
-                .GroupBy(g => new { g.SupportActivityId, date = g.NewRequest.DateRequested })
-                .Select(s => new
-                {
-                    Date = s.Key.date,
-                    Label = ((HelpMyStreet.Utils.Enums.SupportActivities)s.Key.SupportActivityId).FriendlyNameShort(),
-                    Count = s.Count()
-                }).ToList();
-
-            //Get distinct activities from list
-            var supportActivities = chartItems.Select(x => x.Label).Distinct().ToList();
+            return _context.Job
+                    .Include(i => i.NewRequest)
+                    .Where(x => x.NewRequest.ReferringGroupId == groupId && x.NewRequest.DateRequested > minDate)
+                    .Select(s => new DataItem() 
```

Comment:
This is much improved.

I'd just consider going even further and moving the .Select() from the repository to the service, i.e. return jobs from here and create the DataItems higher up the chain.

Same for the other methods below.

Reply:
I try not to return entity objects from the repo layer